### PR TITLE
Add mixup option to ImageNet workloads

### PR DESCRIPTION
**Pull request by runame**
Adds mixup data augmentation option to the ImageNet workloads. It can be selected by specifying the hyperparameters `use_mixup` and optionally `mixup_alpha`. As discussed I went with the less efficient but simpler version where we take the convex combination over the one-hot labels instead over two losses.

### CODE REVIEW

File: algorithmic_efficiency/data_utils.py [hunk 1]
```diff
@@ -28,13 +29,33 @@ def _prepare(x):
   return jax.tree_map(_prepare, xs)
 
 
+def mixup(batch, alpha=0.1):
```
Comment (znado) [inline algorithmic_efficiency/data_utils.py:32]: can you call this `mixup_pytorch`?

File: submission_runner.py [hunk 2]
```diff
@@ -196,12 +196,20 @@ def train_once(
 
   # Workload setup.
   logging.info('Initializing dataset.')
+  use_mixup = (
+      hyperparameters.use_mixup
+      if hasattr(hyperparameters, 'use_mixup') else False)
+  mixup_alpha = (
+      hyperparameters.mixup_alpha
+      if hasattr(hyperparameters, 'mixup_alpha') else 0.1)
```
Comment (znado) [inline submission_runner.py:204]: we will never tune these, because the competition rules prohibit data aug changing on a per-workload basis, so I think we can remove all the changes to submision_runner.py

File: algorithmic_efficiency/spec.py [hunk 3]
```diff
@@ -111,7 +111,9 @@ def build_input_queue(self,
                         global_batch_size: int,
                         cache: Optional[bool] = None,
                         repeat_final_dataset: Optional[bool] = None,
-                        num_batches: Optional[int] = None) -> Dict[str, Any]:
+                        num_batches: Optional[int] = None,
+                        use_mixup: bool = False,
+                        mixup_alpha: float = 0.1) -> Dict[str, Any]:
```
Comment (znado) [inline algorithmic_efficiency/spec.py:116]: (I mention it also below, but) I think we should remove the changes to build_input_queue, because we will always run with a fixed mixup_alpha=0.2 only for imagenet_vit, and submitters aren't allowed to tune it. then we can just pass use_mixup=true only for the training split (see below comments)

File: algorithmic_efficiency/workloads/imagenet_resnet/imagenet_jax/workload.py [hunk 4]
```diff
@@ -30,7 +30,9 @@ def _build_dataset(self,
                      data_dir: str,
                      global_batch_size: int,
                      cache: Optional[bool] = None,
-                     repeat_final_dataset: Optional[bool] = None):
+                     repeat_final_dataset: Optional[bool] = None,
+                     use_mixup: bool = False,
+                     mixup_alpha: float = 0.1):
```
Comment (znado) [inline algorithmic_efficiency/workloads/imagenet_resnet/imagenet_jax/workload.py:35]: we can remove mixup_alpha from here, and then in the imagenet_vit base class, we have a 2 line `_build_dataset` function that just passes `use_mixup=(split == 'train')` to the super class' implementation (same for pytorch)

**Review by znado — APPROVED** [review]
LGTM, I just think we can remove some of the changes to make it even less generic/tunable. Feel free to merge once the comments are resolved.

**Comment by runame** [comment]
Ok makes sense, I have followed your suggestions and restricted mixup to be only used with ImageNet-ViT.
